### PR TITLE
Sort by priority deadline

### LINKED ISSUE #91: Sorting by urgency/importance/status

### PR DESCRIPTION
**Pull request by newway1814**
closes #91 

- [X] Sort tasks by priority.
- [X] Sort tasks by duedate (deadline).

### CODE REVIEW

File: src/main/java/profplan/logic/commands/MarkCommand.java [hunk 1]
```diff
@@ -38,10 +38,12 @@ public MarkCommand(int number) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        if (taskNumber > model.getFilteredTaskList().size()) {
```
Comment (eyelessrhyme7) [inline src/main/java/profplan/logic/commands/MarkCommand.java:41]: Consider moving bug fixes about separate features to separate PRs.

File: src/main/java/profplan/logic/commands/SortDeadlineCommand.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sorts the task by the deadline. "
+            + "Example: " + COMMAND_WORD + " deadline";
+
+    public static final String MESSAGE_SUCCESS = " Here is your task list Prof!, sorted by nearest deadline";
+
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        model.sortTaskByDeadline();
```
Comment (eyelessrhyme7) [inline src/main/java/profplan/logic/commands/SortDeadlineCommand.java:24]: Good job on keeping the methods short and following SLAP!

File: src/main/java/profplan/model/ProfPlan.java [hunk 3]
```diff
@@ -113,7 +113,7 @@ public String toString() {
 
     @Override
     public ObservableList<Task> getTaskList() {
-        return tasks.asUnmodifiableObservableList();
```
Comment (eyelessrhyme7) [inline src/main/java/profplan/model/ProfPlan.java:116]: Can explain this change. I understand you need access to internal list to sort and modify it. Let's make sure it doesn't lead to any unintended behaviour

**Review by eyelessrhyme7 — APPROVED** [review]

File: src/main/java/profplan/model/ProfPlan.java (hunk 3 above)
Comment (newway1814) [reply src/main/java/profplan/model/ProfPlan.java:116]: The list is immutable so could not sort it, that is why had to change it
